lint-staged.config.js

```diff
@@ -13,28 +13,49 @@ function makePattern(extension) {
 module.exports = (stagedFiles) => {
     const eslintExtensions = ['ts', 'tsx', 'js'].map(makePattern);
     const scssExtensions = ['scss'].map(makePattern);
+    const pythonExtensions = ['py'].map(makePattern);
+    const mdExtensions = ['md'].map(makePattern);
     const eslintFiles = micromatch(stagedFiles, eslintExtensions);
     const scssFiles = micromatch(stagedFiles, scssExtensions);
+    const pythonFiles = micromatch(stagedFiles, pythonExtensions);
+    const mdFiles = micromatch(stagedFiles, mdExtensions);
 
-    const tests = containsInPath('/tests/cypress', eslintFiles);
+    const uiTests = containsInPath('/tests/cypress', eslintFiles);
     const cvatData = containsInPath('/cvat-data/', eslintFiles);
     const cvatCore = containsInPath('/cvat-core/src', eslintFiles);
     const cvatCanvas = containsInPath('/cvat-canvas/', eslintFiles);
     const cvatCanvas3d = containsInPath('/cvat-canvas3d/', eslintFiles);
     const cvatUI = containsInPath('/cvat-ui/', eslintFiles);
+    const cvatSdk = containsInPath('/cvat-sdk', pythonFiles);
+    const cvatCli = containsInPath('/cvat-cli', pythonFiles);
+    const pythonTests = containsInPath('/tests/python', pythonFiles);
+
+    const tasks = []; // [command, file list] pairs
+    tasks.push(['npx stylelint --fix ', scssFiles.join(' ')]);
+    tasks.push(['yarn run precommit:cvat-tests -- ', uiTests.join(' ')]);
+    tasks.push(['yarn run precommit:cvat-ui -- ', cvatUI.join(' ')]);
+    tasks.push(['yarn run precommit:cvat-data -- ', cvatData.join(' ')]);
+    tasks.push(['yarn run precommit:cvat-core -- ', cvatCore.join(' ')]);
+    tasks.push(['yarn run precommit:cvat-canvas -- ', cvatCanvas.join(' ')]);
+    tasks.push(['yarn run precommit:cvat-canvas3d -- ', cvatCanvas3d.join(' ')]);
+
+    // If different components use different Black/isort configs,
+    // we need to run Black/isort for each python component group separately.
+    // Otherwise, they all will use the same config. This will mess up
+    // the current package detection for different modules, leading
+    // to incorrect formatting of module imports.
+    tasks.push(['black', cvatSdk.join(' ')]);
+    tasks.push(['black', cvatCli.join(' ')]);
+    tasks.push(['black', pythonTests.join(' ')]);
+    tasks.push(['isort', cvatSdk.join(' ')]);
+    tasks.push(['isort', cvatCli.join(' ')]);
+    tasks.push(['isort', pythonTests.join(' ')]);
+    tasks.push(['pylint', pythonFiles.join(' ')]);
```

Comment:
I need to look into this. This is the same that CI checks, probably some files are still not updated since pylint introduction.

Reply:
I added some files to be ignored by pylint. Migrations are not covered by this, so they will produce warnings when the files are touched. It must be fixed in another PR, but it goes out of the scope for this PR. The problematic migration code itself is, actually, broken, but its dangerous to edit it, because fixes can damage DB.

Reply:
I still see some errors even without counting migrations:

```
************* Module engine.views
cvat/apps/engine/views.py:309:4: W0221: Variadics removed in overridden 'ProjectViewSet.perform_create' method (arguments-differ)
cvat/apps/engine/views.py:817:4: W0221: Variadics removed in overridden 'TaskViewSet.perform_create' method (arguments-differ)
cvat/apps/engine/views.py:1744:4: W0221: Variadics removed in overridden 'IssueViewSet.perform_create' method (arguments-differ)
cvat/apps/engine/views.py:1819:4: W0221: Variadics removed in overridden 'CommentViewSet.perform_create' method (arguments-differ)
************* Module organizations.views
cvat/apps/organizations/views.py:204:4: W0221: Variadics removed in overridden 'InvitationViewSet.perform_create' method (arguments-differ)
************* Module script
utils/dicom_converter/script.py:61:0: W0311: Bad indentation. Found 17 spaces, expected 16 (bad-indentation)
```

> It must be fixed in another PR, but it goes out of the scope for this PR.

I'm fine with _fixing_ migrations in another PR, but we should set them to be ignored until then (and maybe just keep them ignored - I don't think it's reasonable to expect developers to fix up every problem in autogenerated code).

Reply:
The list you mentioned should be covered by #5188.

The migrations are not that obvious. They are generated one time and then they live in the repo, they can be modified, and often they are. In the problematic case the code was modified and clearly has errors.

The strategy of not fixing everything and punishing the developers who touched the files that were not updated since then was accepted quite a long ago. This PR doesn't change anything in this regard. We can raise this topic and discuss it on the next retro.

Reply:
> The list you mentioned should be covered by https://github.com/opencv/cvat/pull/5188.

I think that covers all of them except for `utils/dicom_converter/script.py`.

> The migrations are not that obvious. They are generated one time and then they live in the repo, they can be modified, and often they are. In the problematic case the code was modified and clearly has errors.

Okay, fair enough.

> The strategy of not fixing everything and punishing the developers who touched the files that were not updated since then was accepted quite a long ago. This PR doesn't change anything in this regard. We can raise this topic and discuss it on the next retro.

Okay.